### DISCUSSION TITLE
When will. net 9, C # 13 be supported

### DISCUSSION THREAD

**Discussion by nczsl** [OmniSharp/omnisharp-vscode#7817] [General]
When will. net 9, C # 13 be supported; Now using new features in VSCode will not be recognized. Example:
```csharp
class A<T> where T: allows ref struct{  // red underline
}
```

**Comment by dibarbet** [comment]
C# 13 should be supported in latest version of the extension (and versions before)
![image](https://github.com/user-attachments/assets/e4919bef-4564-4f27-9c28-cee05812ee72)

This is in the Roslyn mode of the extension, I am not sure how much support is present in O# - @JoeRobich ?

**Reply by JoeRobich** [reply]
O# needs an updated Roslyn insertion which has been delayed due to changes in how Roslyn options are managed. This is something I hope to work on over the holidays. Until there is a newer O# release, I would recommend using the Roslyn LSP for projects that require C# 13.
